### PR TITLE
Removed trie state pruning queue for increased performance

### PR DESCRIPTION
**Pull request by iulianpascalau**
## Reasoning behind the pull request
- removed trie state pruning queue
  
## Proposed changes
- removed trie state pruning queue for increased performance

## Testing procedure
- standard system test

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/ElrondNetwork/elrond-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

**Review by BeniaminDrasovean — APPROVED** [review]

**Review by SebastianMarian — APPROVED** [review]

File: integrationTests/state/stateTrie/stateTrie_test.go [hunk 1]
```diff
@@ -1585,6 +1587,75 @@ func TestStatePruningIsNotBuffered(t *testing.T) {
 	}
 }
 
+func TestStatePruningIsNotBufferedOnConsecutiveBlocks(t *testing.T) {
```
Comment (SebastianMarian) [inline integrationTests/state/stateTrie/stateTrie_test.go:1590]: Some golangci linter warns!

File: state/storagePruningManager/evictionWaitingList/memoryEvictionWaitingList.go [hunk 2]
```diff
@@ -130,19 +151,25 @@ func (mewl *memoryEvictionWaitingList) removeFromReversedCache(rootHash []byte,
...
-	hashes, ok := mewl.cache[string(rootHash)]
+	rhData, ok := mewl.cache[string(rootHash)]
 	if !ok {
-		return make(common.ModifiedHashes), nil
+		return make([][]byte, 0), nil
+	}
+
+	if rhData.numReferences <= 1 {
+		delete(mewl.cache, string(rootHash))
+		defer mewl.removeFromReversedCache(rootHash, rhData.hashes)
```
[the remark this answers is not in the document]
Comment (BeniaminDrasovean) [reply state/storagePruningManager/evictionWaitingList/memoryEvictionWaitingList.go:165]: Yes, removed.

**Review by iulianpascalau — COMMENTED** [review]
Looks good from my side :+1:

**Review by SebastianMarian — DISMISSED** [review]

File: state/storagePruningManager/storagePruningManager.go [hunk 3]
```diff
@@ -57,28 +57,46 @@ func (spm *storagePruningManager) MarkForEviction(
...
 	}
+
+	logMapWithTrace("MarkForEviction "+string(identifier), "hash", hashes)
 	return nil
 }
 
+func mapToByteSlice(hashesMap map[string]struct{}) [][]byte {
+	newHashesSlice := make([][]byte, len(hashesMap))
+	i := 0
+	for key := range hashesMap {
```
Comment (AdoAdoAdo) [inline state/storagePruningManager/storagePruningManager.go:92]: is this needed?
can't we work with map[string]struct{}?

File: process/block/baseProcess_test.go [hunk 4]
```diff
@@ -1996,27 +1998,27 @@ func TestBaseProcessor_updateState(t *testing.T) {
...
 			arguments.AccountsDB[state.UserAccountsState],
-			pruningQueue,
 		)
 		prevRootHash = headers[i].RootHash
 
-		if i < numHeaders-1 {
-			assert.Equal(t, 0, len(pruneRootHash))
-			assert.Equal(t, 0, len(cancelPruneRootHash))
+		if i == 0 {
+			assert.Equal(t, []byte("rootHash"), pruneRootHash)
```
[the remark this answers is not in the document]
Comment (BeniaminDrasovean) [reply process/block/baseProcess_test.go:2012]: Done.

File: testscommon/state/evictionWaitingListMock.go [hunk 5]
```diff
@@ -1,40 +1,32 @@
...
-	"github.com/ElrondNetwork/elrond-go-core/data/batch"
-	"github.com/ElrondNetwork/elrond-go-core/marshal"
-	"github.com/ElrondNetwork/elrond-go/common"
 	"github.com/ElrondNetwork/elrond-go/state"
-	"github.com/ElrondNetwork/elrond-go/storage"
 )
 
-// EvictionWaitingList is a structure that caches keys that need to be removed from a certain database.
+// EvictionWaitingListMock is a structure that caches keys that need to be removed from a certain database.
 // If the cache is full, the keys will be stored in the underlying database. Writing at the same key in
```
[the remark this answers is not in the document]
Comment (BeniaminDrasovean) [reply testscommon/state/evictionWaitingListMock.go:12]: Done.

File: state/storagePruningManager/storagePruningManager.go (hunk 3 above)
Comment (BeniaminDrasovean) [reply state/storagePruningManager/storagePruningManager.go:92]: Removed.

**Review by AdoAdoAdo — APPROVED** [review]

**Review by SebastianMarian — APPROVED** [review]

**Review by iulianpascalau — COMMENTED** [review]
Looks good :+1:

**Review by gabi-vuls — COMMENTED** [review]
system test passed

**Review by gabi-vuls — APPROVED** [review]

**Pull request merged**